src/goto-programs/remove_function_pointers.cpp

```diff
@@ -376,30 +382,41 @@ void remove_function_pointerst::remove_function_pointer(
   assert(function.id()==ID_dereference);
   assert(function.operands().size()==1);
 
+  bool found_functions;
+
   const exprt &pointer=function.op0();
   remove_const_function_pointerst::functionst functions;
-  remove_const_function_pointerst fpr(
+  if(does_program_contain_const_removal_cast(goto_program))
+  {
+    warning() << "Cast from const to non-const pointer found, only worst case"
+              << " function pointer removal will be done." << eom;
+    found_functions=false;
+  }
+  else
+  {
+    remove_const_function_pointerst fpr(
     *message_handler, pointer, ns, symbol_table);
```

Comment:
I'm not sure whether github is just hiding whitespace changes at this moment, else the indent needs to be fixed.